### DISCUSSION TITLE
No exception is thrown when `account_id` is not a type of string

### DISCUSSION THREAD

**Discussion by joonas-yoon** [fastapi-users/fastapi-users#1080] [General]
Hi, I am using custom client class for authorizing extends `BaseOAuth2`.

and overrides the method `get_id_email` to get a pair of `account_id`, `account_email`.

The thing that I want to have discussion is, there is no exception when it has wrong type.

for example to be exact, when `get_id_email` returns `(111, 'email@example.com')`, Model can not found by them with Beanie because of the type of it.

```py
async def get_by_oauth_account(
    self, oauth: str, account_id: str
) -> Optional[UP_BEANIE]:
    # ...(skip codes)...
    return await self.user_model.find_one(
        {
            "oauth_accounts.oauth_name": oauth,    # it fails when this is not a string
            "oauth_accounts.account_id": account_id,
        }
    )
```

I know that the methods has "type hints" as like:

```py
async def get_id_email(self, token: str) -> Tuple[str, Optional[str]]:
    ...
```

but as you know, Type hinting do not throw any exceptions.

What do you think about raising exception for this case? or adding assertion for type?

**Comment by joonas-yoon** [comment]
Here is simple solution and it works good.

```diff
# router/oauth.py
def get_oauth_router(...) -> APIRouter:
    async def callback(...):
        # ...(skip)...
        try:
            user = await user_manager.oauth_callback(
                oauth_client.name,
                token["access_token"],
-               account_id,
+               str(account_id),
-               account_email,
+               str(account_email),
                token.get("expires_at"),
                token.get("refresh_token"),
                request,
                associate_by_email=associate_by_email,
            )
        except UserAlreadyExists:
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail=ErrorCode.OAUTH_USER_ALREADY_EXISTS,
            )
```

**Reply by joonas-yoon** [reply]
Oops, this code is in httpx_oauth.

**Comment by frankie567** [comment]
I see what you mean but, as you mentioned above, `get_id_email` should always return a tuple of strings, and take care of casting the `account_id` to a string if necessary.

**Reply by joonas-yoon** [reply]
okay good. I understood. Thanks for reply
